lib/improver/blending/weights.py

```diff
@@ -238,13 +242,478 @@ def process_coord(cube, coordinate, coord_exp_vals=None,
+        axis, = weights_cube.coord_dims(self.weighting_coord_name)
+
+        coord_values = (
+            {self.weighting_coord_name: lambda cell: cell == source_points[0]})
+        constr = iris.Constraint(coord_values=coord_values)
+        lower_fill_value = weights_cube.extract(constr).core_data()
+
+        coord_values = (
+            {self.weighting_coord_name:
+             lambda cell: cell == source_points[-1]})
+        constr = iris.Constraint(coord_values=coord_values)
+        upper_fill_value = weights_cube.extract(constr).core_data()
+
+        fill_value = (lower_fill_value, upper_fill_value)
+        return source_points, target_points, source_weights, axis, fill_value
+
+    def _get_interpolation_inputs_from_dict(self, cube):
+        """
+        Generate inputs required for linear interpolation.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing the coordinate information that will be used
+                for setting up the interpolation inputs.
+
+        Returns:
+            (tuple): tuple containing
+
+                **source_points** (np.ndarray):
+                    Points within the configuration dictionary that will
+                    be used as the input to the interpolation.
+
+                **target_points** (np.ndarray):
+                    Points within the cube that will be the target points
+                    for the interpolation.
+
+                **source_weights** (np.ndarray):
+                    Weights from the configuration dictionary that will be
+                    used as the input to the interpolation.
+
+                **fill_value** (tuple):
+                    Values that be used if extrapolation is required. The
+                    fill values will be used as constants that are extrapolated
+                    if the target_points are outside the source_points
+                    provided. These are equal to the first and last values
+                    provided by the source weights.
+
+        """
+        config_point, = cube.coord(self.config_coord_name).points
+        source_points = (
+            self.config_dict[config_point][self.weighting_coord_name])
+        source_points = np.array(source_points)
+        if "units" in self.config_dict[config_point].keys():
+            units = cf_units.Unit(self.config_dict[config_point]["units"])
+            source_points = units.convert(
+                source_points, cube.coord(self.weighting_coord_name).units)
+
+        target_points = cube.coord(self.weighting_coord_name).points
+        source_weights = (
+            self.config_dict[config_point][self.weights_key_name])
+
+        fill_value = (source_weights[0], source_weights[-1])
+        return source_points, target_points, source_weights, fill_value
+
+    @staticmethod
+    def _interpolate_to_find_weights(
+            source_points, target_points, source_weights, axis=0,
+            fill_value=None):
```

Comment:
Both method seem to provide a fill_value so why is this optional?

Reply:
Fill value must now always be specified.